src/org/pahappa/systems/registrationapp/views/UserView.java

```diff
@@ -133,35 +188,46 @@ private void getUserOfUsername() {
             System.out.println("First Name: " + user.getFirstname());
             System.out.println("Last Name: " + user.getLastname());
             System.out.println("Username: " + user.getUsername());
-            System.out.println("Date Of Birth: " + user.getDateOfBirth());
+            System.out.println("Date Of Birth: " + user.getDateOfBirth().toString().replaceFirst("00:00:00 UTC ", ""));
 
         } catch (Exception e) {
-            System.err.println("An error has been encountered with your inputs. Please hit r to try again.");
-            String retry = scanner.nextLine().strip();
-            if (retry.toLowerCase().equals("r")) {
-                getUserOfUsername();
-            }
+            if (retry()) getUserOfUsername();
         }
     }
 
+    /** returns a string or null */
     private String scanOrNull() {
-        String input = scanner.nextLine().trim();
+        String input = scanner.nextLine().strip();
         if (input == "") return null;
         return input;
     }
 
     private void updateUserOfUsername() {
         try {
             System.out.println("Provide the username below:");
-            String username = scanner.nextLine().trim();
+            String username = spacesCleaner(scanner.nextLine().strip());
+            while (username.length() < MIN_USERNAME_LENGTH || username.length() > MAX_USERNAME_LENGTH) {
+                if (!username.matches("^[a-zA-Z][a-zA-Z0-9_]*")) System.out.println("Invalid characters in username. Usernames start with a letter, and can only allow letters, numbers and underscores.");
+                System.out.println("Username entered must have at least "+MIN_USERNAME_LENGTH+" letters. Please try again: ");
```

Comment:
So if the username statisifies this "while (username.length() < MIN_USERNAME_LENGTH || username.length() > MAX_USERNAME_LENGTH)" but does not satisify this "if (!username.matches("^[a-zA-Z][a-zA-Z0-9_]*")) " then it will still be allowed since it only enters the while loop to check the invalid characters if the length isnt right